### ISSUE TITLE
EPUB3 Multiple Renditions, invalid CSS Media Queries examples

### ISSUE THREAD

**Issue by GoogleCodeExporter** [w3c/epub-specs#498]
```
I believe that the open+close parenthesis pairs are mandatory for the 
"expression" part(s) of the Media Query syntax

http://www.w3.org/TR/css3-mediaqueries/#syntax

http://www.idpf.org/epub/renditions/multiple/#h.v5umqueir7kw


```

Original issue reported on code.google.com by `daniel.weck` on 24 Feb 2015 at 12:08

**Comment by danielweck** [comment]
Originally posted by @danielweck
(now subscribed)

**Comment by mattgarrish** [comment]
The media type is also required to be valid, but I'm trying to configure validation to allow it to not be specified.

I can live without it, since no other value can be specified, but making the parens optional seems destined to lead to malformed media queries being reported as valid, at least without additional complexity to verify that all parentheses are properly balanced when they are used.

**Comment by murata2makoto** [comment]
First, I believe that "(" and ")" should be required.

Second, I suppose that we would like to allow the logical operation
"and".

Third, our current text disallow HTML/CSS media types other than
"all".  (Note: HTML/CSS media types of media queries should not be
confused with MIME media types).  This restriction is reasonable,
since we have rendition:accessMode.

But we not disallow "not all", which is useless.  We do not disallow

  all and (min-width: 1920px),  all and (max-width: 3000px)

either.  We might simply disallow the use of HTML/CSS media types
including "all".

The BNF in CSS media queries is:

media_query_list
 : S\* [media_query [ ',' S\* media_query ]\* ]?
 ;
media_query
 : [ONLY | NOT]? S\* media_type S\* [ AND S\* expression ]*
 | expression [ AND S\* expression ]*

expression
 : '(' S\* media_feature S\* [ ':' S\* expr ]? ')' S*

If we disallow the use of HTML/CSS media types, the first line of the
second rule will not be used.

This means that we can specify 

  rendition:media="(min-width: 1920px)"

  rendition:media="(min-width: 1920px) and (max-width: 3000px)"

  rendition:media="(color), (min-width: 1920px)"

The last one means either color screen or width >= 1920px.

In summary, here is my proposal.

   A media query list without a HTML/CSS media type, where the syntax of a
   media query list is specified by the non-terminal media_query_list
   in Clauss 3 of CSS 3 media query [MediaQueries].

Examples:

  rendition:media="(min-width: 1920px)"

  rendition:media="(min-width: 1920px) and (max-width: 3000px)"

  rendition:media="(color), (min-width: 1920px)"

  rendition:media="all and (color)"

The last example is incorrect, since it specifies an HTML/CSS media 
type "all".

**Comment by mattgarrish** [comment]
I agree, FWIW. We excluded all the other media types, which makes including “all” in the actual media query redundant, especially when we also say it is implied.

I’d rather we make it implied for all cases and remove the ability to include an explicit value, as you’ve suggested.

**Comment by murata2makoto** [comment]
I hope to close this issue in the teleconf.

**Comment by danielweck** [comment]
Regarding @murata0204 's proposal to drop support for `media_type` entirely:
http://www.w3.org/TR/css3-mediaqueries/#syntax
The first line of the `media_query` production rule would be removed:
`: [ONLY | NOT]? S* media_type S* [ AND S* expression ]*`
-- I am okay with this proposed restriction to the standard CSS Media Queries syntax. --
**UPDATE**: the `ONLY` and `NOT` keywords apply to both the `media_type` and (optional) `expressions` as a whole, not just to the `media_type`. I therefore retract my "okay" statement above.

**Comment by mattgarrish** [comment]
Since we lose the ability to write only/not expressions if we strip the media type, I'd recommend we stay with the prose we have, which allows the media_type to be specified but requires it be "all".

**Comment by mattgarrish** [comment]
Note from the Media Queries specification:

The logical NOT can be expressed through the ‘not’ keyword. The presence of the keyword ‘not’ at the beginning of the media query negates the result. I.e., if the media query had been true without the ‘not’ keyword it will become false, and vice versa.

So not/only don't just apply to the media type but to the expression as a whole. I'm worried what problems we'd introduce by not allowing them. Without real-world use, it's not easy to say.

**Comment by danielweck** [comment]
Indeed, the `ONLY` and `NOT` keywords apply to both the `media_type` and (optional) `expressions` as a whole, not just to the `media_type`. I therefore retract my "okay" statement above (which was in support of removing / disabling the first line of the `media_query` production rule).
As @mattgarrish said, the `NOT` keyword **must** be used in combination with the `media_type`, so we can preserve the current specification prose as it is (`all` is the only value allowed for `media_type`).
See:
https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Media_queries#not

**Comment by mgylling** [comment]
Ah, great catch. My fault for misleading you on the call. +1 to the current prose.

**Comment by mattgarrish** [comment]
Okay, I'm closing leaving that prose unchanged but making the other fixes. Feel free to re-open if anyone disagrees.
